src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_CompoundAssignmentOperator.cs

```diff
@@ -388,16 +393,16 @@ private BoundIndexerAccess TransformIndexerAccess(BoundIndexerAccess indexerAcce
         /// Returns true if the <paramref name="receiver"/> was lowered and transformed.
         /// The <paramref name="receiver"/> is not changed if this function returns false. 
         /// </summary>
-        private bool TransformCompoundAssignmentFieldOrEventAccessReceiver(Symbol fieldOrEvent, ref BoundExpression receiver, ArrayBuilder<BoundExpression> stores, ArrayBuilder<LocalSymbol> temps)
+        private bool TransformCompoundAssignmentFieldOrEventAccessReceiver(Symbol fieldOrEvent, ref BoundExpression? receiver, ArrayBuilder<BoundExpression> stores, ArrayBuilder<LocalSymbol> temps)
         {
             Debug.Assert(fieldOrEvent.Kind == SymbolKind.Field || fieldOrEvent.Kind == SymbolKind.Event);
 
             //If the receiver is static or is the receiver is of kind "Base" or "this", then we can just generate field = field + value
-            if (fieldOrEvent.IsStatic || !CanChangeValueBetweenReads(receiver))
+            if (fieldOrEvent.IsStatic || !CanChangeValueBetweenReads(receiver!))
```

Comment:
>receiver! [](start = 69, length = 9)

I didn't understand this suppression. Do you want to put [DisallowNull] on the `receiver` parameter?

Reply:
No.  Only when the symbol is not static.  That's why this appears after the `||`.   I'll move it to an assertion.

---
In reply to: [386624759](https://github.com/dotnet/roslyn/pull/42068#discussion_r386624759) [](ancestors = 386624759)